synthetic_data/dataset_generators/datetime_generator.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+import pandas as pd
+from numpy.random import Generator
+from typing import Optional
+
+
+def generate_datetime(
```

Comment:
I would recommend putting the default values for the start and end date variables in function's doc string, just because if we ever want to autogenerate documentation it would be good to have in there already